test/integration-tests/sdk-test-suites/cpp/testsuite_sdk.cpp

```diff
@@ -46,37 +47,26 @@ class TestSuiteSDK : public TestSuiteTalent {
     }
 };
 
-static auto mqtt_config = json{
-    {"platform", true},
-    {"module", {
-                   {"name", "./iotea-sdk-cpp-lib/adapters/mqtt/libmqtt_protocol_adapter.so"}
-               }
-    },
-    {"config",
-        {
-            {"brokerUrl", "tcp://mosquitto:1883"},
-            {"topicNamespace", "iotea/"}
-        }
-    }
-};
-static auto gateway_config = ProtocolGateway::CreateConfig(json{mqtt_config});
-static auto gateway = std::make_shared<ProtocolGateway>(gateway_config);
-static Client client{gateway};
+std::shared_ptr<Client> client;
 
 void signal_handler(int) {
-    std::cout << "Stopping..." << std::endl;
-    client.Stop();
+    if (client) {
+        client->Stop();
+    }
 }
 
-
 int main(int, char**) {
-    auto talent = std::make_shared<TestSuiteSDK>();
+    std::ifstream file{"../../../config/tests/cpp/config.json"};
+    std::string config{std::istreambuf_iterator<char>(file), std::istreambuf_iterator<char>()};
 
-    client.RegisterFunctionTalent(talent);
+    auto gateway = std::make_shared<ProtocolGateway>(json::parse(config));
+    client = std::make_shared<Client>(gateway);
 
-    std::signal(SIGINT, signal_handler);
+    auto talent = std::make_shared<TestSuiteSDK>();
+    client->RegisterTalent(talent);
 
-    client.Start();
+    std::signal(SIGINT, signal_handler);
```

Comment:
SIGINT is for Ctrl-C, but e.g. edgecontainerd is sending SIGTERM when you ask it to stop the container.
Should it handle both signals?

Reply:
I have added SIGTERM as well